parser/modify.go

```diff
@@ -14,47 +14,50 @@
 
 package parser
 
-func AddStringToList(value *Value, s string) (modified bool) {
-	if value.Type != List {
-		panic("expected list value, got " + value.Type.String())
+import "fmt"
+
+func AddStringToList(value Expression, s string) (modified bool) {
```

Comment:
Why not take a List here instead of an Expression? (and in RemoveStringFromList)